MSAL/test/integration/native_auth/end_to_end/sign_up/MSALNativeAuthSignUpUsernameAndPasswordEndToEndTests.swift

```diff
@@ -68,22 +72,31 @@ final class MSALNativeAuthSignUpUsernameAndPasswordEndToEndTests: MSALNativeAuth
         signUpVerifyCodeDelegate.signInAfterSignUpState?.signIn(delegate: signInAfterSignUpDelegate)
 
         await fulfillment(of: [signInExp])
-        checkSignInAfterSignUpDelegate(signInAfterSignUpDelegate)
+        checkSignInAfterSignUpDelegate(signInAfterSignUpDelegate, expectedUsername: username)
     }
 
-    // Hero Scenario 2.1.2. Sign up - with Email verification as LAST step & Custom Attributes (Email & Password)
+    // Hero Scenario 1.1.3. Sign up - with Email verification as LAST step & Custom Attributes (Email & Password)
     func test_signUpWithPassword_withEmailVerificationAsLastStepAndCustomAttributes_succeeds() async throws {
-        throw XCTSkip("Skipping this test because native auth KeyVault is missing")
-        guard let sut = initialisePublicClientApplication() else {
+        guard
+            let sut = initialisePublicClientApplication(clientIdType: .passwordAndAttributes),
+            let password = await retrievePasswordForSignInUsername()
```

Comment:
instead of retrieving the signIn password, can you just generate a random valid password? There is no need to access the key vault for those tests

Reply:
Thanks, it makes a lot of sense. I've created a `generateRandomPassword()` function in `MSALNativeAuthEndToEndBaseTestCase` with the [SSPR's technique](https://github.com/AzureAD/microsoft-authentication-library-for-objc/pull/2259/files) to generate random passwords.

In this way it can also be used by the SSPR code.